### ISSUE TITLE
Can not add this package to Xcode 16

### ISSUE THREAD

**Issue by sebsto** [aws-geospatial/amazon-location-mobile-auth-sdk-ios#12]
When trying to add this package to a new project in Xcode 16, I receive the following error 

![image](https://github.com/user-attachments/assets/75eb03d0-ceed-42a9-8cde-623fda3c2d00)

**Comment by cgalvan** [comment]
@sebsto Hi, can you please paste the full error message for this issue? We are investigating to reproduce locally.

Does it match something like the following?

```
Failed to resolve dependencies Dependencies could not be resolved because package 'amazon-location-mobile-tracking-sdk-ios' is required using a stable-version but 'amazon-location-mobile-tracking-sdk-ios' depends on an unstable-version package 'amazon-location-mobile-auth-sdk-ios' and root depends on 'amazon-location-mobile-tracking-sdk-ios' 0.2.1.
```

**Comment by cgalvan** [comment]
We were able to reproduce the issue with Xcode 15 as well, so it is not specific to XCode 16. There is a workaround to specify the package version as `Branch` instead of `Up to Next Major Version`, but we are still investigating the reason why specifying it by version is failing to resolve. Will update on this issue once we know more.

**Comment by sebsto** [comment]
Using branch = main works. But the order is important when adding the packages.

I added first the tracker library (main branch). This downloads the Auth library as a dependency (v0.2.2). 
If I try to install the auth library afterwards (as instructed in [the doc](https://docs.aws.amazon.com/location/latest/developerguide/qs-create-resources-ios.html)), it fails with the error message shared above.

**Comment by mbalfour-amzn** [comment]
@sebsto We've just released v0.3.1 of the Auth library that should fix the error you reported. Thanks for making us aware of the issue!

**Comment by sebsto** [comment]
I can confirm that we can now add version 0.3.1 to an Xcode 16 project - thank you
